src/app/core/services/notifications/notification-wrapper.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core'
+import { Platform } from 'ionic-angular'
+
+import { DefaultNotificationType } from '../../../../assets/data/defaultConfig'
+import { ConfigKeys } from '../../../shared/enums/config'
+import { NotificationMessagingType } from '../../../shared/models/notification-handler'
+import { RemoteConfigService } from '../config/remote-config.service'
+import { StorageService } from '../storage/storage.service'
+import { AppServerRestNotificationService } from './app-server-rest-notification.service'
+import { FcmXmppNotificationService } from './fcm-xmpp-notification.service'
+import { LocalNotificationService } from './local-notification.service'
+import { NotificationService } from './notification.service'
+
+@Injectable()
+export class NotificationWrapperService extends NotificationService {
```

Comment:
this is basically a factory. You could call it NotificationServiceFactory and use the factory get the instance of NotificationService in other classes.